test/gtest/common/mem_buffer.cc

```diff
@@ -221,7 +221,7 @@ void mem_buffer::get_bar1_free_size_nvml()
 #endif
 }
 
-void *mem_buffer::allocate(size_t size, ucs_memory_type_t mem_type)
+void *mem_buffer::allocate(size_t size, ucs_memory_type_t mem_type, bool async)
```

Comment:
Adding `async` argument may look misleading, because it can be passed for any memory type.
Maybe we can use extended enum for memory type?
```C
typedef enum {
    MEMORY_TYPE_CUDA_ASYNC = UCS_MEMORY_TYPE_LAST
} ucs_memory_type_ext_t;
```
And use this enum in allocate and release methods

Reply:
> Adding `async` argument may look misleading, because it can be passed for any memory type. Maybe we can use extended enum for memory type?
> 
> ```c
> typedef enum {
>     MEMORY_TYPE_CUDA_ASYNC = UCS_MEMORY_TYPE_LAST
> } ucs_memory_type_ext_t;
> ```
> 
> And use this enum in allocate and release methods

I had the idea in mind. However, it would lead to significant code changes for the sake of a rather special use case.